packages/arb-avm-cpp/data_storage/include/data_storage/checkpointstorage.hpp

```diff
@@ -33,20 +36,23 @@ class TransactionDB;
 class CheckpointStorage {
```

Comment:
once codes become mutable, is the plan to have to make new storages and keep track of older ones as well?

Reply:
That's a great question. My current thought is that we'll use shared memory and mutexes to handle things. It's made a bit easier by the fact that code is immutable, it can just be added to. Potentially we'll want machines that are adding to their code to commit to the shared store when they get cloned or checkpointed. This is definitely something important to figure out.

Given that the approach this replaces is basically infeasible memory-wise I think it makes sense to do this now and then figure out how to handle mutable code when we get there.